src/utils/generateKeyInsightsData.js

```diff
@@ -0,0 +1,70 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+import {
+  ADMISSIONS_REVOCATIONS_PAROLE,
+  ADMISSIONS_REVOCATIONS_PROBATION,
+  POPULATION_PRISON,
+} from "../constants/metrics";
+
+const generateKeyInsightsData = (flowData) => {
+  const keyInsights = [];
+
+  if (!flowData[POPULATION_PRISON].isNotAvailable) {
+    const isPositive = flowData[POPULATION_PRISON].numberChange > 0;
+
+    keyInsights.push({
+      ...flowData[POPULATION_PRISON],
+      caption: `The prison population ${isPositive ? "rose" : "fell"} ${Math.abs(
+        flowData[POPULATION_PRISON].percent
+      )}, ${isPositive ? "an increase" : "a decline"} of ${Math.abs(
+        flowData[POPULATION_PRISON].numberChange
+      )} people.`,
+    });
+  }
+
+  if (!flowData[ADMISSIONS_REVOCATIONS_PAROLE].isNotAvailable) {
+    const isPositive = flowData[ADMISSIONS_REVOCATIONS_PAROLE].numberChange > 0;
+
+    keyInsights.push({
```

Comment:
It looks like the formatting logic in here is fairly duplicative, especially between parole revocations and probation revocations, which are identical except for one word in the text and a key into the `flowData` object. Can we consolidate it?